requirements.txt

```diff
@@ -1,50 +1,12 @@
-aiomysql==0.2.0
-aiosqlite==0.19.0
-annotated-types==0.5.0
-anyio==3.7.1
-asyncpg==0.28.0
-boto3==1.28.63
-certifi==2023.7.22
-charset-normalizer==3.2.0
-chromadb==0.4.11
-click==8.1.7
-exceptiongroup==1.1.3
-faiss-cpu==1.7.4
-fastapi==0.99.1
-flatten-json==0.1.13
-gdown==4.7.1
-greenlet==2.0.2
-h11==0.14.0
-idna==3.4
-iniconfig==2.0.0
-nest-asyncio==1.5.7
-networkx==3.1
-numba==0.58.0
-numpy==1.24.4
-packaging==23.1
-pandas==2.0.3
-Pillow==10.1.0
-pluggy==1.3.0
-pydantic==1.10.12
-PyMySQL==1.1.0
-pytest==7.4.2
-python-dateutil==2.8.2
-pytz==2023.3.post1
-PyYAML==6.0.1
-requests==2.31.0
-scipy==1.10.1
-six==1.16.0
-sniffio==1.3.0
+nest-asyncio>=1.5.7
```

Comment:
See above to check if there's a way to consolidate these somehow...

Reply:
Hi @ddkang , we can consolidate these required dependencies with the extra optional dependencies into a single requirements file while still maintaining the on-demand installation behavior. Can you please confirm if that is what we need here?

Reply:
@continue-revolution @ttt-77 can you check this?

Reply:
@shubhankarunhale You can do that and check if it works for Python 3.12

Reply:
Okay, sure. Checking if it works with Python 3.12 for compatible libraries